Document WebXR layer init & XRCompositionLayer (#8948)
mdn/content: files/en-us/web/api/xrcompositionlayer/needsredraw/index.md

---
title: XRCompositionLayer.needsRedraw
slug: Web/API/XRCompositionLayer/needsRedraw
tags:
  - API
  - Property
  - Reference
  - VR
  - WebXR
  - WebXR API
  - WebXR Device API
  - XR

browser-compat: api.XRCompositionLayer.needsRedraw
---
{{APIRef("WebXR Device API")}}

The read-only **`needsRedraw`** property of the {{domxref("XRCompositionLayer")}} interface is a boolean signaling that the layer should be rerendered in the next frame.

The need for redrawing can occur when the underlying resources of a layer are lost due to textures that might got freed by the compositor under the hood. This might happen when the device falls asleep or when the browser switches context (to an operating system dialog or similar) and then comes back.

> Redrawing is not a problem for layers which are updated with every frame. However, for layers updated infrequently, or for static layers (where you can only draw once after creation or after a `redraw` event), the layer's content might be lost and needs to get redrawn. If the layer's resources are lost, it is indicated by `needsRedraw` property becoming `true` and a `redraw` event is fired on the layer.

Reviewer: ```suggestion
Redrawing is not a problem for layers which are updated with every frame. However, for layers updated infrequently, or for static layers (where you can only draw once after creation or after a `redraw` event), the layer's content might be lost and need to be redrawn. If the layer's resources are lost, the `needsRedraw` property will be `true` and a `redraw` event is fired on the layer.
```

## Value

> A boolean. `true` indicates a re-rendering is needed in the next frame, `false` indicates no re-rendering is needed.

Reviewer: ```suggestion
A boolean. `true` indicates a rerendering is needed in the next frame, `false` indicates no re-rendering is needed.
```

Reviewer: For words that aren't in the dictionary I grep the repo to see which is more common. This one came out 42/41. The dictionary documents what's done, not what should be done (contrary to popular belief). Google trends shows [the no-hyphen version winning](https://trends.google.com/trends/explore?date=all&geo=US&q=rerender,re-render). Also, the no-hyphen version is clear enough that that the hyphen doesn't aid comprehension.

Author: I'm fine with either. Accepted your suggestion.
